File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        temp=n
        summ=0
        prod=1
        total=0
        while(temp>0):
            dig=temp%10
            summ+=dig
            prod*=dig
            temp//=10
        total=summ+prod
        if(n%total==0):
            return True
        return False

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int temp=n;
        int sum=0;
        int prod=1;
        int total=0;
        while(temp>0)
        {
            int dig=temp%10;
            sum+=dig;
            prod*=dig;
            temp/=10;
        }
        total=sum+prod;
        if(n%total==0)
        {
            return true;
        }
        return false;
    }
}